add user dashboard

nitinkushwaha17 opened the pull request.

nitinkushwaha17 committed:
add user dashboard

nitinkushwaha17 committed:
merge

SantamRC commented on src/Components/userDashboard/Dashboard.jsx:
```diff
@@ -0,0 +1,240 @@
+    <BrowserRouter>
+    <div className={classes.root}>
+      <CssBaseline />
+      <AppBar position="absolute" className={clsx(classes.appBar, open && classes.appBarShift)}>
```
Remove the App Bar and keep the original Navbar as it is in the site.

SantamRC commented on src/Components/userDashboard/Dashboard.jsx:
```diff
@@ -0,0 +1,240 @@
+import LayersIcon from '@material-ui/icons/Layers';
+import AssignmentIcon from '@material-ui/icons/Assignment';
+
+const drawerWidth = 240;
```
Can this be put inside the function body?

nitinkushwaha17 committed:
remove appbar from user dashboard

nitinkushwaha17 committed:
remove appbar from user dashboard

nitinkushwaha17 committed:
make drawer responsive

nitinkushwaha17 committed:
make drawer responsive

nitinkushwaha17 committed:
minor change

nitinkushwaha17 committed:
minor change

nitinkushwaha17 committed:
add research statement page

nitinkushwaha17 committed:
add research statement page

SantamRC commented on src/Components/userDashboard/ResearchStatement.jsx:
```diff
@@ -0,0 +1,117 @@
+        fontSize: '20px',
+        padding: '10px',
+    },
+    '@media screen and (max-width: 992px)': {
+        btnContainer: {
+            width: '90%'
+        },
+        txt:{
+            width: '90%'
+        },
+        txtArea: {
+            width: '90%'
+        }
+    }
```
Use the theme breakpoints feature given by Material UI, like here https://material-ui.com/customization/breakpoints/

nitinkushwaha17 committed:
use theme breakpoints

nitinkushwaha17 committed:
use theme breakpoints

SantamRC commented on src/Components/userDashboard/ResearchStatement.jsx:
```diff
@@ -0,0 +1,117 @@
+    );
+}
+
+function Statement(props) {
+    const classes = useStyles();
+
+    let [text, setText] = useState(localStorage.getItem('statement')||props.text);
+
+    function handleChange(e){
+        setText(e.target.value)
+    }
+
+    return(
+        props.ie?
+        <textarea value={text} maxLength={1500} onChange={handleChange} className={classes.txtArea} />:
+        <pre className={classes.txt}>{props.text}</pre>
+    );
+}
```
Make a separate jsx file inside this folder named Statement.jsx and then import it into this file
Do the same for project card

nitinkushwaha17 committed:
separate components

nitinkushwaha17 committed:
separate components

nitinkushwaha17 committed:
make drawer sticky

nitinkushwaha17 committed:
make drawer sticky

nitinkushwaha17 committed:
add announcements page

nitinkushwaha17 committed:
add announcements page

nitinkushwaha17 committed:
Merge branch 'master' of https://github.com/nitinkushwaha17/ResearchPortalFrontend

nitinkushwaha17 committed:
changes

nitinkushwaha17 committed:
changes

nitinkushwaha17 committed:
changes

nitinkushwaha17 committed:
run prettier

nitinkushwaha17 committed:
merge

nitinkushwaha17 committed:
Merge branch 'ieeesb-nitdgp-master'

nitinkushwaha17 committed:
merge

nitinkushwaha17 committed:
fix eslint errors

nitinkushwaha17 committed:
test

nitinkushwaha17 committed:
improve drawer

SantamRC commented:
Some things to be changed:
In the dashboard page-
- Apply for more button will take the user to the dashboard map
- Map the project bodies using the map function from the projects.json file in Data and display each fields
- Add a delete button also in the projects
- Change ongoing projects to "Projects Applied"

nitinkushwaha17 committed:
merge

nitinkushwaha17 committed:
merge

nitinkushwaha17 committed:
merge

nitinkushwaha17 committed:
eslintcache

nitinkushwaha17 committed:
improve main dashboard card

nitinkushwaha17 committed:
CRLF to LF

nitinkushwaha17 committed:
run prettier

nitinkushwaha17 committed:
impovements

1. improved project card ui and added delete button

2. linked main dashboard button to projects route

3. renamed main dashboard title

4. removed resume tab and route

5. removed old dashboard files and route

nitinkushwaha17 committed:
add profile page

nitinkushwaha17 committed:
Merge branch 'master' into master

nitinkushwaha17 committed:
add resume section in profile

nitinkushwaha17 committed:
Merge branch 'ieeesb-nitdgp:master' into master

nitinkushwaha17 committed:
Merge branch 'master' of https://github.com/nitinkushwaha17/ResearchPortalFrontend

nitinkushwaha17 committed:
minor improvements

nitinkushwaha17 committed:
add bottom-nav
